constraint/pkg/client/crd_helpers.go

```diff
@@ -118,8 +142,8 @@ func (h *crdHelper) validateCR(cr *unstructured.Unstructured, crd *apiextensions
 	if cr.GroupVersionKind().Group != constraintGroup {
 		return fmt.Errorf("Wrong group for constraint %s. Have %s, want %s", cr.GetName(), cr.GroupVersionKind().Group, constraintGroup)
 	}
-	if cr.GroupVersionKind().Version != crd.Spec.Version {
-		return fmt.Errorf("Wrong version for constraint %s. Have %s, want %s", cr.GetName(), cr.GroupVersionKind().Version, crd.Spec.Version)
+	if !supportedVersions[cr.GroupVersionKind().Version] {
```

Comment:
Shouldn't this check against the supported versions of `constraints.gatekeeper.sh`not `templates.gatekeeper.sh`?

Reply:
I'm assuming these will be 1:1, but that's not necessarily guaranteed.

Reply:
This is probably ok for now, but we might need to come back to this later.

Reply:
ack, LMK if you'd much rather prefer I add some constants somewhere. I don't really have a preference personally.